torchvision/utils.py

```diff
@@ -397,42 +397,51 @@ def flow_to_image(flow: torch.Tensor) -> torch.Tensor:
     Converts a flow to an RGB image.
 
     Args:
-        flow (Tensor): Flow of shape (2, H, W) and dtype torch.float.
+        flow (Tensor): Flow of shape (N, 2, H, W) or (2, H, W) and dtype torch.float.
 
     Returns:
-        img (Tensor(3, H, W)): Image Tensor of dtype uint8 where each color corresponds to a given flow direction.
+        img (Tensor): Image Tensor of dtype uint8 where each color corresponds
+            to a given flow direction. Shape is (N, 3, H, W) or (3, H, W) depending on the input.
     """
 
     if flow.dtype != torch.float:
         raise ValueError(f"Flow should be of dtype torch.float, got {flow.dtype}.")
 
-    if flow.ndim != 3 or flow.size(0) != 2:
-        raise ValueError(f"Input flow should have shape (2, H, W), got {flow.shape}.")
+    orig_shape = flow.shape
+    if flow.ndim == 3:
+        flow = flow[None]  # Add batch dim
```

Comment:
One small doubt.

Does this modify flow inplace?

Reply:
No, the 2 objects may be a view on the same underlying memory, but they're different objects


```py
In [1]: import torch
torch.tensor
In [2]: a = torch.tensor([1, 2])

In [3]: b = a[None]

In [4]: b is a
Out[4]: False

In [5]: a, b
Out[5]: (tensor([1, 2]), tensor([[1, 2]]))
```

Reply:
Thanks for clarification. Will note this.